src/main/java/de/numcodex/feasibility_gui_backend/query/broker/dsf/DSFFhirWebClientProvider.java

```diff
@@ -113,6 +117,37 @@ public WebsocketClient provideFhirWebsocketClient() throws FhirWebClientProvisio
                 proxyContext.getUsername(),
                 proxyContext.getPassword(),
                 null, subscriptionId);
+        wsRef.set(client);
+        return client;
+    }
+
+    /**
+     * Package-visible helper that constructs the reconnector Runnable. Extracted for easier testing.
+     *
+     * @param wsRef an AtomicReference holding the WebsocketClient instance used for reconnect attempts
+     * @return a Runnable that starts a background thread to attempt reconnection
+     */
+    static Runnable createReconnector(AtomicReference<WebsocketClient> wsRef) {
+        return () -> {
+            new Thread(() -> {
```

Comment:
Creating a new thread is most likely an antipattern. ForkJoinPool.commonPool().submit() should be preffered.